test/phoenix_live_view/html_tokenizer_test.exs

```diff
@@ -64,11 +65,75 @@ defmodule Phoenix.LiveView.HTMLTokenizerTest do
 
       assert [
                {:tag_open, "p", [], %{line: 1, column: 1}},
-               {:text, "\n<!--\n<div>\n-->\n", %{line_end: 5, column_end: 1}},
+               {:text, "\n<!--\n<div>\n-->\n",
+                %{line_end: 5, column_end: 1, context: [:comment_start, :comment_end]}},
                {:tag_close, "p", %{line: 5, column: 1}},
                {:tag_open, "br", [], %{line: 5, column: 5}}
              ] = tokenize(code)
     end
+
+    test "adds comment_start and comment_end" do
+      first_part = """
+      <p>
+      <!--
+      <div>
+      """
+
+      {tokens, cont} = HTMLTokenizer.tokenize(first_part, "nofile", 0, [], [], :text)
+
+      assert Enum.reverse(tokens) ==
+               [
+                 {:tag_open, "p", [], %{column: 1, line: 1}},
+                 {:text, "\n<!--\n<div>\n",
+                  %{column_end: 1, line_end: 4, context: [:comment_start]}}
+               ]
+
+      second_part = """
+      </div>
+      -->
+      </p>
+      """
+
+      {tokens, :text} = HTMLTokenizer.tokenize(second_part, "nofile", 0, [], tokens, cont)
+
+      assert Enum.reverse(tokens) == [
+               {:tag_open, "p", [], %{column: 1, line: 1}},
+               {:text, "\n<!--\n<div>\n",
+                %{column_end: 1, line_end: 4, context: [:comment_start]}},
+               {:text, "</div>\n-->\n", %{column_end: 1, context: [:comment_end], line_end: 3}},
+               {:tag_close, "p", %{column: 1, line: 3}},
+               {:text, "\n", %{column_end: 1, line_end: 4}}
+             ]
+    end
+
+    test "handle comment and with script" do
```

Comment:
here I'm trying to reproduce the scenario where parse comment + eex + script as you pointed out.  Here is how it is tokenized:

```elixir
iex(> EEx.Tokenizer.tokenize(contents, 0, 0, %{indentation: 0, trim: false})
{:ok,
 [
   {:text, 0, 0, '<!-- '},
   {:expr, 0, 5, '=', ' "Hello" '},
   {:text, 0, 19, ' --><script>'},
   {:expr, 0, 31, '=', ' "World" '},
   {:text, 0, 45, '</script>'},
   {:eof, 0, 54}
 ]}
```

Because we append eex expressions into comments and script there in the Formatter, that is exactly how it would be processed here in the `HTMLTokenizer`.

Also, notice that different from comments, script has tag open and close. So I believe we don't need any metadata for them.

Reply:
The specific details are not relevant IMO. The important is that you can achieve what you need in the formatter. :)

Reply:
Yep. I will drop this test, it was just to exemplify that case you mentioned and help me to understand what would happen :)